### PR TITLE
refacto, typos, 496b -> 481b, fix dependecies for benchmark script

### PR DESCRIPTION
**Pull request by B2o5T**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,3 @@
-dist
-node_modules
\ No newline at end of file
+dist/
+node_modules/
+.idea/
```
Comment (B2o5T) [inline .gitignore:3]: for WebStorm users

File: package.json [hunk 2]
```diff
@@ -20,7 +20,8 @@
     "build": "yarn build:mjs && yarn build:cjs"
   },
   "devDependencies": {
-    "esbuild": "^0.12.28"
+    "esbuild": "^0.12.28",
+    "prettier": "^2.7.1"
```
Comment (B2o5T) [inline package.json:24]: prettier was declared in `package.json` but missing in devDeps

File: src/index.ts [hunk 3]
```diff
@@ -1,7 +1,7 @@
 // Use WeakMap to store the object-key mapping so the objects can still be
 // garbage collected. WeakMap uses a hashtable under the hood, so the lookup
 // complexity is almost O(1).
-const table = new WeakMap<object, number | string>()
+const table = new WeakMap<object, string>()
```
Comment (B2o5T) [inline src/index.ts:4]: I guess it's a mistake as you save only `string`s

File: src/index.ts [hunk 4]
```diff
@@ -13,55 +13,44 @@ let counter = 0
...
     if (constructor == Array) {
       // Array.
       result = "@"
       for (index = 0; index < arg.length; index++) {
         result += stableHash(arg[index]) + ","
       }
-      table.set(arg, result)
-    }
-    if (constructor == Object) {
+    } else if (constructor == Object) {
```
Comment (B2o5T) [inline src/index.ts:35]: `else if` in this case is better

File: src/index.ts [hunk 5]
```diff
@@ -13,55 +13,44 @@ let counter = 0
...
       // Object, sort keys.
       result = "#"
       const keys = Object.keys(arg).sort()
-      while (typeof (index = keys.pop() as string) !== "undefined") {
-        if (typeof arg[index] !== "undefined") {
+      while ((index = keys.pop() as string) !== undefined) {
+        if (arg[index] !== undefined) {
```
Comment (B2o5T) [inline src/index.ts:40]: same behaviour, less bundle

File: src/index.ts [hunk 6]
```diff
@@ -13,55 +13,44 @@ let counter = 0
...
     // If it's already hashed, directly return the result.
-    result = table.get(arg)
+    let result = table.get(arg)
     if (result) return result
-
-    // Store the hash first for circular reference detection before entering the
-    // recursive `stableHash` calls.
-    // For other objects like set and map, we use this id directly as the hash.
-    result = ++counter + "~"
-    table.set(arg, result)
```
Comment (shuding) [inline src/index.ts:35]: This is necessary, otherwise recursive objects won't work:

```
const a = []
a[0] = a

stableHash(a) // stack overflow
```

**Review by shuding — COMMENTED** [review]
Wow thanks, I think I missed this notification

File: src/index.ts (hunk 6 above)
Comment (B2o5T) [reply src/index.ts:35]: @shuding Hey! I guess it's hard to make a review for these changes, but I didn't change original logic of code
image

I only replaced highlighted lines by single one before `return result`

**Review by shuding — APPROVED** [review]

**Comment by shuding** [comment]
Thank you! I have some other ideas to make it even smaller (but not that necessary haha).

**Pull request merged**